### PR TITLE
[draf] unify drop catalog and metalake

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

 - add `in-use` property to metalake and catalog
 - add `in-use` to drop catalog/metalake APIs
 - add `inactivate()` and `activate()` API for metalake and catalog
 - new definition for drop matalake/catalog

### Why are the changes needed?

unify drop catalog and metalake

### Does this PR introduce _any_ user-facing change?

yes

### How was this patch tested?

tests added

### CODE REVIEW

**Comment by mchades** [comment]
@jerryshao Could you please review the new API definition when you have time? Thank you.

File: api/src/main/java/org/apache/gravitino/SupportsCatalogs.java [hunk 1]
```diff
@@ -108,12 +110,67 @@ Catalog alterCatalog(String catalogName, CatalogChange... changes)
...
+   * </ul>
+   *
+   * It is equivalent to calling {@code dropCatalog(ident, false)}.
+   *
+   * @param ident the identifier of the catalog.
+   * @return True if the catalog was dropped, false if the catalog does not exist.
+   * @throws NonEmptyEntityException If the catalog is not empty.
+   * @throws EntityInUseException If the catalog is in use.
+   */
+  boolean dropCatalog(NameIdentifier ident) throws NonEmptyEntityException, EntityInUseException;
```
Comment (jerryshao) [inline api/src/main/java/org/apache/gravitino/SupportsCatalogs.java:129]: We should use String catalog name, not catalog name identifier.

File: api/src/main/java/org/apache/gravitino/SupportsCatalogs.java [hunk 2]
```diff
@@ -108,12 +110,67 @@ Catalog alterCatalog(String catalogName, CatalogChange... changes)
...
+   *       org.apache.gravitino.exceptions.NonInUseEntityException}.
+   *   <li>Any operation on the sub-entities (schemas, tables, etc.) will throw an {@link
+   *       org.apache.gravitino.exceptions.NonInUseEntityException}.
+   * </ul>
+   *
+   * @param ident The identifier of the catalog.
+   * @throws NoSuchCatalogException If the catalog does not exist.
    */
-  boolean dropCatalog(String catalogName);
+  void inactivateCatalog(NameIdentifier ident) throws NoSuchCatalogException;
```
Comment (jerryshao) [inline api/src/main/java/org/apache/gravitino/SupportsCatalogs.java:173]: deactivate, not inactivate.

**Comment by jerryshao** [comment]
I need to think more about it, generally the API design LGTM.

**Comment by mchades** [comment]
I will close this PR, the rest of the tasks will be tracked by #5031

**Pull request closed without merging**